frappe/integrations/doctype/google_drive/google_drive.py

```diff
@@ -0,0 +1,241 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019, Frappe Technologies and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+import requests
+import googleapiclient.discovery
+import google.oauth2.credentials
+import os
+
+from frappe import _
+from googleapiclient.errors import HttpError
+from frappe.model.document import Document
+from frappe.utils import get_request_site_address
+from six.moves.urllib.parse import quote
+from apiclient.http import MediaFileUpload
+from frappe.utils.file_manager import save_file, get_file_url
+from frappe.utils.print_format import download_pdf
+from frappe.utils import get_backups_path, get_files_path
+from frappe.utils.backups import new_backup
+from frappe.utils import now
+
+SCOPES = "https://www.googleapis.com/auth/drive"
+
+class GoogleDrive(Document):
+
+	def validate(self):
+		if self.enable_system_backup:
+			system_backup = frappe.db.exists("Google Drive", {"enable_system_backup": 1})
+			if system_backup and not system_backup == self.name:
+				frappe.throw(_("Google Drive System Backup can be enabled only for one account."))
+
+	def get_access_token(self):
+		google_settings = frappe.get_doc("Google Settings")
+
+		if not google_settings.enable:
+			frappe.throw(_("Google Integration is disabled."))
+
+		if not self.refresh_token:
+			button_label = frappe.bold(_("Allow Google Drive Access"))
+			raise frappe.ValidationError(_("Click on {0} to generate Refresh Token.").format(button_label))
+
+		data = {
+			"client_id": google_settings.client_id,
+			"client_secret": google_settings.get_password(fieldname="client_secret", raise_exception=False),
+			"refresh_token": self.get_password(fieldname="refresh_token", raise_exception=False),
+			"grant_type": "refresh_token",
+			"scope": SCOPES
+		}
+
+		try:
+			r = requests.post("https://www.googleapis.com/oauth2/v4/token", data=data).json()
```

Comment:
maybe it would be a better idea to create a global variable for this? seems like you're using this in multiple places. would be easier to replace if google makes any api changes